controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -669,23 +319,37 @@ func remove(slice []string, s string) (result []string) {
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *StorageConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	enqueueStorageConsumerRequest := handler.EnqueueRequestsFromMapFunc(
+		func(client client.Object) []reconcile.Request {
+			return []reconcile.Request{{
+				NamespacedName: types.NamespacedName{
+					Name:      "StorageConsumer",
```

Comment:
I never did this before, But I think we should not hardcode this name and this name should not be static, there can be multiple storage consumers, and we may want to enqueue the different one every time.

Reply:
Yes, we need to only watch the storage consumer that is governing this specific claim
We can check the owner ref on the claim and get the name of the consumer, but I think it is bad practice to load a resource inside a predicate (even the claim itself) 

What is available to us inside the predicate is the claim name, namespace, labels, and annotations.
We can have an annotation on the claim specifying the consumer this claim belongs to and use it to watch the correct storage consumer.

@subhamkrai can we make this change?

Reply:
Adding an example of a predicate that uses the object labels to filter 
https://github.com/red-hat-storage/ocs-osd-deployer/blob/main/controllers/managedocs_controller.go#L229-L236

Reply:
we already have storageConsumerUiD in storageCLassClaim, I can do that to get the claim name? does this work?

Reply:
or instead of the name can we just use the uid of consumer for the enqueue? @iamniting @nb-ohad

Reply:
You don't have the entire StorageClassClaim available for you inside the predicate, only a portion of the metadata. 
And you should not run `get` operation inside predicates as well (the predicate sig cannot even return an error).

If UID is available for you to filter with, then yes, please use it.

Reply:
@nb-ohad I think we are not on the same page, As far as I understand we are watching a block-pool change and would like to reque the storage consumer.

storage consumer does not have a direct relation with the block-pool, So we can not get a storage consumer name from a block-pool directly.

So we have to figure out the storage consumer in the below sequence.

block-pool (storage class claim is an owner to it) -> storage class claim (storage consumer is an owner to it, we also have a label) -> storage consumer

So in any case I think we have to do a get on the storage class claim if I am not mistaken.

Reply:
I can think of one possible solution at this point, We can also add the label or annotation on the block pool while creating it with a storage consumer name as the storage class claim is going to create a block pool that does have a knowledge of the storage consumer.

Reply:
Or we can add another owner ref on the blockpool for the consumer and use `Own(` watch instead of watch + predicates. But we will create another issue around block pool deletion ( garbage collection)

Reply:
I like the label method much more than owner ref as that will help us everywhere to even filter out the block pools later. I think we should have 2 labels on the block pool one for the storage class claim and one for the storage consumer. It will definitely help us in the longer run.

Reply:
I also like the label idea since it will not bring any issues and also helps us to filter

Reply:
Changes looks good to me now